webui/src/Buttons/EditButton.tsx

```diff
@@ -372,6 +373,59 @@ interface TabsSectionProps {
 	feedbacks: FeedbackInstance[]
 }
 
+interface TabStepProps{
+	selectedKey: string | false
+	k: string | number
+	i: number
+	name:string | undefined
+	inputState: {showInputEle:boolean}
+	linkClassname: string | undefined
+	socket: CompanionSocketType
+	controlId: string
+
+}
+function TabStep({selectedKey, k, i, name, inputState, linkClassname, socket, controlId }: TabStepProps){
+
+	const renameStep = useCallback(
+		(stepId: string, newName: string) => {
+			socketEmitPromise(socket, 'controls:step:rename', [controlId, stepId, newName]).catch((e) => {
+				console.error('Failed to rename step:', e)
+			})
+		},
+		[socket, controlId]
+	)
+	
+	function onChangeHandler(e: React.ChangeEvent<HTMLInputElement>){
+		setState({showInputEle: state.showInputEle});
+		if(selectedKey){
+			renameStep(selectedKey, e.target.value);
+		}
+	}
+
+	const [state, setState] = useState(inputState);
+	return (
+		<CNavItem key={k} className="nav-steps-special">
+			{state.showInputEle? (
+				<CNavLink className={linkClassname}>
+				<input
+					type="text"
+					value={name}
+					onChange={(e)=>{onChangeHandler(e)}}
+					onKeyDown={(e)=>{e.key === 'Enter' && setState({showInputEle: false})}}
+					onBlur={(e)=>setState({showInputEle: false})}
+					autoFocus
+				>			
+				</input>
+				</CNavLink>
+			) : (
+				<CNavLink onDoubleClick={(e)=>setState({showInputEle: true})} data-tab={`step:${k}`} className={linkClassname}>
+					{name && name !== "" ? name : (i === 0 ? 'Step ' + (i + 1) : i + 1)}
```

Comment:
Would it make sense to do `name (${i + 1})` when there is a name set?
I suspect that if you have something like 8 named steps, figuring out which is step 4 will be tedious without us keeping the number at the end